File: Data Structures & Algorithms/subsets/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    helper(nums,i,current,result){
        if(i>=nums.length){
            result.push([...current])
            return
        }
        this.helper(nums,i+1,current,result)
        current.push(nums[i])
        this.helper(nums,i+1,current,result)
        current.pop()
    }
    subsets(nums) {
        const current=[],result=[]
        this.helper(nums,0,current,result)
        return result
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        L,R,result=0,len(nums)-1,nums[0]
        while L<=R:
          if nums[L]<=nums[R]:
            result=min(result,nums[L])
            break
          mid=(L+R)//2
          if nums[mid]>=nums[L]:
            result=min(result,nums[L])
            L=mid+1
          elif nums[mid]<nums[L]:
            result=min(result,nums[mid])
            R=mid
        return result
